src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,46 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_NUMBERS_OF_ELEMENTS = 10;
+    private Pair[] pairs;
+    private int size;
+
+    public StorageImpl() {
+        pairs = new Pair[MAX_NUMBERS_OF_ELEMENTS];
+    }
+
     @Override
     public void put(K key, V value) {
+        for (int i = 0; i < size; i++) {
+            if (checkKey(key) != null) {
+                pairs[i].setValue(value);
+                return;
+            }
+        }
+        Pair<K, V> pair = new Pair<>();
+        pair.setKey(key);
+        pair.setValue(value);
+        pairs[size] = pair;
+        size++;
```

Comment:
```suggestion
        Pair<K, V> existingPair = getPair(key);
        if (existingPair != null) {
            existingPair.setValue(value);
            return;
        }
        Pair<K, V> pair = new Pair<>();
        pair.setKey(key);
        pair.setValue(value);
        pairs[size] = pair;
        size++;
```
метод checkKey(key) довольно трудоемкий, поэтому его не стоит вызывать в цикле(если это возможно)